web/client/reducers/maplayout.js

```diff
@@ -6,9 +6,20 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import { UPDATE_MAP_LAYOUT } from '../actions/maplayout';
+import {UPDATE_DOCK_PANELS, UPDATE_MAP_LAYOUT} from '../actions/maplayout';
 
 import assign from 'object-assign';
+import ConfigUtils from "../utils/ConfigUtils";
+import {get} from "lodash";
+
+export const DEFAULT_RIGHT_DOCK_PANELS = ['mapCatalog', 'mapTemplates', 'metadataexplorer', 'userExtensions', 'details'];
+export const DEFAULT_LEFT_DOCK_PANELS = [];
+
+const customRightPanels = get(ConfigUtils.getConfigProp('miscSettings'), 'exclusiveDockPanels.right', []);
+const customLeftPanels = get(ConfigUtils.getConfigProp('miscSettings'), 'exclusiveDockPanels.left', []);
+
+const rightDockPanels = DEFAULT_RIGHT_DOCK_PANELS.concat(...(Array.isArray(customRightPanels) ? customRightPanels : []));
+const leftDockPanels = DEFAULT_LEFT_DOCK_PANELS.concat(...(Array.isArray(customLeftPanels) ? customLeftPanels : []));
```

Comment:
we should not export additional element from a reducers. The reducer should only export the correspondent function. If we need those constant accessible we should move those in a utils. Do we need those two constant available in other files?

```suggestion
const DEFAULT_RIGHT_DOCK_PANELS = ['mapCatalog', 'mapTemplates', 'metadataexplorer', 'userExtensions', 'details'];
const DEFAULT_LEFT_DOCK_PANELS = [];
```

Below the reason related to the proposed changes

In my opinion we should avoid to mix global configuration with initial state. MapStore provides an initialState property in the localConfig where is possible to configure the defaultState and we could do this for example:

```js
"initialState": {
  "defaultState": {
      ...
      "maplayout": {
        "dockPanels": {
          "left": [], // custom list
          "right": [] // custom list
        }
```

The snippet above will replace the inital state for the maplayout including the list of panels needed in a specific project.

My proposals are:

- remove additional getConfigProp 'exclusiveDockPanels' if the initialState.defaultState property is enough
- if initialState.defaultState is not enough we could think of an action to update dockPanels property. We are using state here so I think the best option would be to use an action
- I would avoid to concat new panels in the project but instead replace completely the list if the project needs custom panels. This will avoid to add specific property to exclude/include panel to the list